src/mlpack/core/tree/rectangle_tree/minimal_coverage_sweep.hpp

```diff
@@ -0,0 +1,118 @@
+/**
+ * @file minimal_coverage_sweep.hpp
+ * @author Mikhail Lozhnikov
+ *
+ * Definition of the MinimalCoverageSweep class, a class that finds a partition
+ * of a node along an axis.
+ */
+#ifndef MLPACK_CORE_TREE_RECTANGLE_TREE_MINIMAL_COVERAGE_SWEEP_HPP
+#define MLPACK_CORE_TREE_RECTANGLE_TREE_MINIMAL_COVERAGE_SWEEP_HPP
+
+namespace mlpack {
+namespace tree {
+
+constexpr double fillFactor = 0.5;
+
+/**
+ * The MinimalCoverageSweep class finds a partition along which we
+ * can split a node according to the coverage of two resulting nodes.
+ * Moreover, the class evaluates the cost of each split.
+ *
+ * @tparam SplitPolicy The class that provides rules for inserting children of
+ *    a node that is being split into two new subtrees.
+ */
+template<typename SplitPolicy>
+class MinimalCoverageSweep
+{
+ private:
+  /**
+   * Class to allow for faster sorting.
+   */
+  template<typename ElemType>
+  struct SortStruct
+  {
+    ElemType d;
+    int n;
+  };
```

Comment:
I think that `SortStruct` is used a few other places, but does it give us any advantage over `std::pair<ElemType, int>`?  It's possible that `SortStruct` is faster, but I am not sure of that.  Have you played with that at all?